pkg/controller/network/securitygroup/controller.go

```diff
@@ -152,54 +144,54 @@ func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.Ex
 	cr.Status.SetConditions(runtimev1alpha1.Creating())
 
 	// Creating the SecurityGroup itself
-	req := e.client.CreateSecurityGroupRequest(&awsec2.CreateSecurityGroupInput{
+	req := e.sg.CreateSecurityGroupRequest(&awsec2.CreateSecurityGroupInput{
 		GroupName:   aws.String(cr.Spec.GroupName),
-		VpcId:       aws.String(cr.Spec.VPCID),
+		VpcId:       cr.Spec.VPCID,
 		Description: aws.String(cr.Spec.Description),
 	})
 
 	result, err := req.Send(ctx)
 	if err != nil {
 		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
 	}
-
 	cr.UpdateExternalStatus(awsec2.SecurityGroup{GroupId: result.GroupId})
-
-	// Authorizing Ingress permissions for the SecurityGroup
-	ingressPerms := v1alpha3.BuildEC2Permissions(cr.Spec.IngressPermissions)
-	if len(ingressPerms) > 0 {
-		air := e.client.AuthorizeSecurityGroupIngressRequest(&awsec2.AuthorizeSecurityGroupIngressInput{
-			GroupId:       aws.String(cr.Status.SecurityGroupID),
-			IpPermissions: ingressPerms,
-		})
-
-		_, err = air.Send(ctx)
-		if err != nil {
-			return managed.ExternalCreation{}, errors.Wrap(err, errAuthorizeIngress)
-		}
-	}
-
-	// Authorizing Egress permissions for the SecurityGroup
-	egressPerms := v1alpha3.BuildEC2Permissions(cr.Spec.EgressPermissions)
-	if len(egressPerms) > 0 {
-		aer := e.client.AuthorizeSecurityGroupEgressRequest(&awsec2.AuthorizeSecurityGroupEgressInput{
-			GroupId:       aws.String(cr.Status.SecurityGroupID),
-			IpPermissions: egressPerms,
-		})
-
-		_, err = aer.Send(ctx)
-		if err != nil {
-			return managed.ExternalCreation{}, errors.Wrap(err, errAuthorizeEgress)
-		}
+	// We need to save status before spec update so that it's not lost.
+	if err := e.kube.Status().Update(ctx, cr); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errStatusUpdate)
 	}
 
-	return managed.ExternalCreation{}, nil
+	// NOTE(muvaf): GroupID is used as external name instead of GroupName because
+	// there are cases where only GroupID is accepted as identifier.
+	meta.SetExternalName(cr, aws.StringValue(result.GroupId))
+	return managed.ExternalCreation{}, errors.Wrap(e.kube.Update(ctx, cr), errSpecUpdate)
 }
 
 func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
```

Comment:
Just want to verify that if we add `Ingress` and `Egress` rules in `Update` rather than `Create` that the `SecurityGroup` will be created "locked down" by default? Basically I just don't want us to be creating a more permissive security group than the user requests and then updating it. Which I guess could be possible if the user requests something that is more restrictive than the default 🤔 I believe that the default allows inbounf traffic from other instances in the SG, and all outbound traffic https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-security-groups.html#default-security-group

Reply:
The section you linked describes the default SG that you get when you create a `VPC`. I interpreted the next section as what we are concerned; [what rules are default when I create a custom security group](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-security-groups.html#creating-your-own-security-groups). It seems like it doesn't allow any inbound traffic but allows all outbound traffic. I feel like this is restrictive enough for a default, what do you think?

Reply:
I can see @hasheddan's concern, but isn't this inherent to the AWS API? It seems like the API requires us to first create the group then modify its rules (otherwise we'd just do it as one create API call). So the question here is more about _how long_ the security group will be using the default set of rules before we update it. Previously we did both in the same `Create` call, and now we'll do it in `Update`. Keep in mind this probably doesn't delay the calls very much, because we'll be immediately requeued after a creation, notice we need an update, and apply the missing rules.

Reply:
I went ahead and debugged this using breakpoints. What's happening is that there is no inbound rules and only one outbound rule by default which allows `0.0.0.0`. Even though their API is designed that way, what we do is only **addition**. So, there isn't any insecure->secure switch between `Create`->`Update` for inbound rules. For outbound, it's the same as before; we get default `0.0.0.0` rule and leave it there because we don't know about it and API is additive.